onnx/shape_inference/implementation.cc

```diff
@@ -303,10 +302,12 @@ static void InferShapesImpl(
     throw std::runtime_error(inference_errors);
   }
     for (auto& vi : *g->mutable_output()) {
-    auto iter = outputTypesByName.find(vi.name());
-    if (iter != outputTypesByName.end() && iter->second != NULL) {
-      vi.mutable_type()->mutable_tensor_type()->set_elem_type(iter->second);
-    }
+      if (!vi.name().empty()) {
+        auto iter = outputTypesByName.find(vi.name());
+        if (iter != outputTypesByName.end() && iter->second != NULL) {
+          vi.mutable_type()->mutable_tensor_type()->set_elem_type(iter->second);
```

Comment:
The basic idea here is good. However, the existing code has another limitation, which shows up here too: it assumes that the inputs/outputs of the graph are all tensor-types. It would be nice to update this to allow it to be of other types (like sequence, etc.) also.

Reply:
But may be we should make it a TODO comment and address it separately, rather than let this PR become too big. I don't think it is affecting any existing models currently.